validate.js

```diff
@@ -1,22 +1,35 @@
 'use strict';
-var util = require('util'),
-fs = require('fs'),
-jpath = require('json-path'),
-path = require('path');
-
-
-function getValue(obj, key, transFn){
-	if((0 == key.indexOf('/') || 0 == key.indexOf('#/')) && transFn) {
-		return jpath.resolve(obj, key)
-	}
-	return obj[key]
+const util = require('util');
+const fs = require('fs');
+const path = require('path');
+const { JSONPath } = require('jsonpath-plus');
+
+function getValue(obj, key, transFn) {
+  if (key.startsWith('$') && transFn) {
```

Comment:
This is a very different conditional than the original.
"Starts with `/` or `#/`"
vs.
"Starts with `$`"

Was this intentional? If so, then why wasn't that done with the `hasKey` function below?

Reply:
FYI this was covered in point 1 of the PR description (re: xpath becomes jsonpath)